### PR TITLE
udev: Updates for cdrom_id helper

### PR DESCRIPTION
**Pull request by pali**
[cdrom_id](https://github.com/systemd/systemd/blob/master/src/udev/cdrom_id/cdrom_id.c) udev helper does not parse all MMC profiles. Following change fixes this issue and parse all 34 profiles from all MMC standard versions. Also it replaces magic constants by macros provided by linux/cdrom.h and fixes cd_profiles_old_mmc() to issue READ_DISC_INFO command in two steps, like it is doing kernel and also mkudffs.

### CODE REVIEW

**Comment by yuwata** [comment]
Thanks.

File: src/udev/cdrom_id/cdrom_id.c [hunk 1]
```diff
@@ -479,14 +676,14 @@ static int cd_profiles(int fd) {
...
-        scsi_cmd_set(&sc, 0, 0x46);
+        scsi_cmd_set(&sc, 0, GPCMD_GET_CONFIGURATION);
         scsi_cmd_set(&sc, 8, 8);
         scsi_cmd_set(&sc, 9, 0);
         err = scsi_cmd_run(&sc, fd, features, 8);
         if (err != 0) {
                 info_scsi_cmd_err("GET CONFIGURATION", err);
                 /* handle pre-MMC2 drives which do not support GET CONFIGURATION */
-                if (SK(err) == 0x5 && IN_SET(ASC(err), 0x20, 0x24)) {
+                if (SK(err) == 0x5 && (ASC(err) == 0x20 || ASC(err) == 0x24)) {
```
Comment (yuwata) [inline src/udev/cdrom_id/cdrom_id.c]: Why did you change this line? I prefer to use `IN_SET()` macro.

**Review by yuwata — COMMENTED** [review]
LGTM. One minor point.

**Comment by yuwata** [comment]
Thank you for your quick update!

File: src/udev/cdrom_id/cdrom_id.c [hunk 2]
```diff
@@ -26,49 +27,85 @@
...
 static unsigned cd_media_hddvd_r;
+static unsigned cd_media_hddvd_r_dl;
 static unsigned cd_media_hddvd_rw;
+static unsigned cd_media_hddvd_rw_dl;
 static unsigned cd_media_mo;
+static unsigned cd_media_mo_se; /* sector erase */
+static unsigned cd_media_mo_wo; /* write once */
+static unsigned cd_media_mo_as; /* advance storage */
 static unsigned cd_media_mrw;
 static unsigned cd_media_mrw_w;
```
Comment (poettering) [inline src/udev/cdrom_id/cdrom_id.c:112]: hmm, so thee are all booleans? given you touch it maybe make them all C99 bools now. It's how we have been doing for new code in the rest of the codebase, this is just some old leftover. But when you touch this, please update it.

That said, maybe a flags field might actually be better here, given how many those are now

File: src/udev/cdrom_id/cdrom_id.c [hunk 3]
```diff
@@ -987,14 +1248,22 @@ int main(int argc, char *argv[]) {
...
         if (cd_media_hddvd)
                 printf("ID_CDROM_MEDIA_HDDVD=1\n");
         if (cd_media_hddvd_r)
                 printf("ID_CDROM_MEDIA_HDDVD_R=1\n");
+        if (cd_media_hddvd_r_dl)
+                printf("ID_CDROM_MEDIA_HDDVD_R_DL=1\n");
         if (cd_media_hddvd_rw)
                 printf("ID_CDROM_MEDIA_HDDVD_RW=1\n");
+        if (cd_media_hddvd_rw_dl)
+                printf("ID_CDROM_MEDIA_HDDVD_RW_DL=1\n");
```
Comment (poettering) [inline src/udev/cdrom_id/cdrom_id.c:1281]: given how large this already was, and how much larger this will get now, can we please rework this to be a table? an array of structs mapping the values to properties to set?

File: src/udev/cdrom_id/cdrom_id.c (hunk 2 above)
Comment (pali) [reply src/udev/cdrom_id/cdrom_id.c:112]: My change is using same coding style and same approach as existing code.

So this looks like a technical debt in your project which is not relevant to the issue which I'm trying to fix.

I do not have a time nor capacity to remove this technical debt in your project.

Comment (yuwata) [reply src/udev/cdrom_id/cdrom_id.c:112]: OK, let's do that coding style change later.
